source/_docs/guides/jenkins.md

```diff
@@ -0,0 +1,310 @@
+
+    ![Credentials Page](/source/docs/assets/images/integrations/credentials.png)
+
+2. In a new browser tab, go to your GitHub account. Go to [Settings](https://github.com/settings/profile). Under **Developer Settings** click on **Personal access tokens**  and generate a new token. This should have all **repo**  and **admin:repo-hook** options:
+
+    ![GitHub token permissions](/source/docs/assets/images/integrations/jenkins-gh-token-access.png)
+
+    Copy the generated token. Be careful, as you will not be able to view it again.
+
+
+3. Back in Jenkins, create your new credential with the following options:
+
+    - **Kind**: Secret Text
+    - **Scope**: Global
+    - **Secret**: Paste the GitHub token you created.
+    - **ID**: blank
+    - **Description**: GitHub
+
+    Press OK to add the new credential to the system.
+
+4. Create another credential for a Terminus Token, using  either an existing token or [create a new token](/docs/machine-tokens) from your Jenkins user dashboard:
+
+    - **Kind**: Secret Text
+    - **Scope**: Global
+    - **Secret**: Paste the Terminus token you created.
+    - **ID**: blank
+    - **Description**: Terminus
+
+    After saving both, your credentials will be accessible for secure use.
+
+    ![Credentials Page](/source/docs/assets/images/integrations/2_credentials.png)
+
+    <div class="alert alert-info" role="alert">
+    <h4 class="info">Note</h4>
+    <p markdown="1">Keep your Terminus token handy, you'll need to enter it again later.</p>
+    </div>
+
+5. Return to the main Jenkins dashboard and select **Manage Jenkins**, then **Configure System**.
+
+6. Scroll to the **GitHub Servers** section  and click on **Add GitHub Server**.
+
+7. Leave the **API URL** as `https://api.github.com`. For **Credentials**, click the drown-down menu and select your GitHub credentials.
+
+8. Click **Test Connection** and you should see the GitHub username which created the token.
+
+9. Click **Save**.
+
+
+### Create the Jenkins Project
+
+1. Log into the Jenkins dashboard as an admin user. Click on **New Item**.
+
+    ![Jenkins dashboard](/source/docs/assets/images/integrations/jenkins_dash.png)
+
+2. Give the project a name with no spaces. Select **Freestyle Project** and click **OK** to save.
+
+    ![New Jenkins project](/source/docs/assets/images/integrations/new_job.png)
+
+3. The next page lets you configure options for this project. In the **General** tab, select "GitHub project" and enter the repository URL (e.g. `https://github.com/YOUR-ORG/YOUR-PROJECT`).
+
+4. In the **Source Code Management** tab select **Git**, then **Advanced**. Add the following information:
+
+    - **Repository URL**: The path to the .git file of your repository, e.g. `https://github.com/YOUR-ORG/YOUR-PROJECT.git`
+    - **Name**: `origin`
+    - **Refspec**: blank
+    - **Branch specifier**: `origin/*`
+    - **Additional Behaviours**: Add **Prune stale remote-tracking branches**:
+
```

Comment:
Can we explain what some of these configurations do and how they affect CI behavior?